web/client-api/src/main/java/io/deephaven/web/client/api/JsPartitionedTable.java

```diff
@@ -95,19 +92,25 @@ public Promise<JsPartitionedTable> refetch() {
                     WebBarrageUtils.readSchemaMessage(descriptor.getConstituentDefinitionSchema_asU8()));
             ColumnDefinition[] columnDefinitions = tableDefinition.getColumns();
             Column[] columns = new Column[0];
-            Column[] keyColumns = new Column[0];
             for (int i = 0; i < columnDefinitions.length; i++) {
                 ColumnDefinition columnDefinition = columnDefinitions[i];
                 Column column = columnDefinition.makeJsColumn(columns.length, tableDefinition.getColumnsByName());
                 columns[columns.length] = column;
-                if (descriptor.getKeyColumnNamesList().indexOf(columnDefinition.getName()) != -1) {
-                    keyColumnTypes.add(columnDefinition.getType());
-                    keyColumns[keyColumns.length] = column;
+            }
+            Column[] keyColumns = new Column[0];
+            JsArray<String> keyColumnNames = descriptor.getKeyColumnNamesList();
+            for (int i = 0; i < keyColumnNames.length; i++) {
+                String name = keyColumnNames.getAt(i);
+                ColumnDefinition columnDefinition = tableDefinition.getColumnsByName().get(false).get(name);
+                int index = 0;
+                while (!columns[index].getName().equals(name)) {
+                    index++;
                 }
+                keyColumnTypes.add(columnDefinition.getType());
+                keyColumns[keyColumns.length] = columns[index];
```

Comment:
No need to loop twice, i think this would do the job too?
```suggestion
                ColumnDefinition columnDefinition = tableDefinition.getColumnsByName().get(false).get(name);
                keyColumnTypes.add(columnDefinition.getType());
                keyColumns[keyColumns.length] = columns[columnDefinition.getIndex()];
```